packages/virtual-core/src/index.ts

```diff
@@ -646,6 +646,10 @@ export class Virtualizer<
     },
     {
       key: false,
+      onChange: () => {
+        // Notify when measurement options change as they affect total size
+        this.notify(false)
+      },
```

Comment:
@shunn2 I created a PR earlier (https://github.com/TanStack/virtual/pull/1088) that adds an initialization guard to the memo system so we avoid the extra initial `notify` call. This keeps the `onChange` semantics consistent (only firing when state actually changes), preserves the existing React re-render count expectations, and still fixes the stale `getTotalSize` issue.

With that now merged into `master`, you can simplify this block to

```suggestion
      skipInitialOnChange: true
      onChange: () => {
        // Notify when measurement options change as they affect total size
        this.notify(this.isScrolling)
      },
```